src/libsync/propagatedownload.h

```diff
@@ -51,6 +49,10 @@ class OWNCLOUDSYNC_EXPORT GETFileJob : public AbstractNetworkJob
     /// Will be set to true once we've seen a 2xx response header
     bool _saveBodyToFile = false;
 
+protected:
+    QIODevice *_device;
```

Comment:
I understand why you have made it protected but that will make our life more complicated once we want to refactor `GETFileJob` because we have to check the whole code of `GETFileJob` and `GETEncryptedFileJob`

I would suggest that `_decryptor->chunkDecryption` returns the decrypted data and we use `GETFileJob::writeToDevice` to write to the device no longer needing access within `GETEncryptedFileJob`